src/roblox/instance/terrain.rs

```diff
@@ -0,0 +1,127 @@
+use mlua::prelude::*;
+use rbx_dom_weak::types::{MaterialColors, TerrainMaterials, Variant};
+
+use crate::roblox::{
+    datatypes::types::{Color3, EnumItem},
+    shared::classes::{add_class_restricted_method, add_class_restricted_method_mut},
+};
+
+use super::Instance;
+
+pub const CLASS_NAME: &str = "Terrain";
+
+fn material_from_name(material_name: &str) -> Option<TerrainMaterials> {
+    match material_name {
+        "Grass" => Some(TerrainMaterials::Grass),
+        "Slate" => Some(TerrainMaterials::Slate),
+        "Concrete" => Some(TerrainMaterials::Concrete),
+        "Brick" => Some(TerrainMaterials::Brick),
+        "Sand" => Some(TerrainMaterials::Sand),
+        "WoodPlanks" => Some(TerrainMaterials::WoodPlanks),
+        "Rock" => Some(TerrainMaterials::Rock),
+        "Glacier" => Some(TerrainMaterials::Glacier),
+        "Snow" => Some(TerrainMaterials::Snow),
+        "Sandstone" => Some(TerrainMaterials::Sandstone),
+        "Mud" => Some(TerrainMaterials::Mud),
+        "Basalt" => Some(TerrainMaterials::Basalt),
+        "Ground" => Some(TerrainMaterials::Ground),
+        "CrackedLava" => Some(TerrainMaterials::CrackedLava),
+        "Asphalt" => Some(TerrainMaterials::Asphalt),
+        "Cobblestone" => Some(TerrainMaterials::Cobblestone),
+        "Ice" => Some(TerrainMaterials::Ice),
+        "LeafyGrass" => Some(TerrainMaterials::LeafyGrass),
+        "Salt" => Some(TerrainMaterials::Salt),
+        "Limestone" => Some(TerrainMaterials::Limestone),
+        "Pavement" => Some(TerrainMaterials::Pavement),
+        _ => None,
+    }
+}
```

Comment:
This sucks... I think rbx-dom needs an `impl FromStr for TerrainMaterials`

Reply:
That would be nice, but I think this is definitely fine for now